test/library/standard/OS/posix_signals.chpl

```diff
@@ -0,0 +1,28 @@
+use OS.POSIX;
+
+writeln(
+    SIGABRT !=
+    SIGALRM !=
+    SIGBUS !=
+    SIGCHLD !=
+    SIGCONT !=
+    SIGFPE !=
+    SIGHUP !=
+    SIGILL !=
+    SIGINT !=
+    SIGKILL !=
+    SIGPIPE !=
+    SIGQUIT !=
+    SIGSEGV !=
+    SIGSTOP !=
+    SIGTERM !=
+    SIGTRAP !=
+    SIGTSTP !=
+    SIGTTIN !=
+    SIGTTOU !=
+    SIGURG !=
+    SIGUSR1 !=
+    SIGUSR2 !=
+    SIGXCPU !=
+    SIGXFSZ
```

Comment:
Doesn't this boolean expression have the effect of:

* comparing two signals, determining that they're not the same and concluding the result is "true"/effectively 1
* then comparing the next signal (and all following) to `1` rather than the signal value?

I think to get the effect I'm guessing you're looking for you'd need to do an O(n**2) comparison of all signal values to see that none of them are the same.

Reply:
Or cast them to integer and put them in a set and then count the number of set elements

Reply:
That's a much smarter approach.